CHANGELOG.md

```diff
@@ -12,6 +12,379 @@ The format is inspired by [Keep a Changelog].
+  This is a breaking runtime host protocol change as the version types in
+  protocol are changed.
+
+- runtime: Add support for block tags
+  ([#3848](https://github.com/oasisprotocol/oasis-core/issues/3848))
+
+- runtime: Add epoch to ExecuteTx/CheckTx/Query requests
+  ([#3892](https://github.com/oasisprotocol/oasis-core/issues/3892))
+
+- go/cmd/registry: simplify register runtime command
+  ([#3903](https://github.com/oasisprotocol/oasis-core/issues/3903))
+
+  The `registry runtime gen_register` command now accepts a JSON runtime
+  descriptor, which should simplify generating runtime registration
+  transactions.
+  The `registy runtime init_genesis` command is removed as runtime
+  descriptors in genesis are not singed since version 21.0+ making the command
+  obsolete.
+
+- staking: Include share amounts in `AddEscrow` and `ReclaimEscrow` events
+  ([#3939](https://github.com/oasisprotocol/oasis-core/issues/3939))
+
+- Update emitted events when disbursing rewards
+  ([#3944](https://github.com/oasisprotocol/oasis-core/issues/3944))
+
+  Before, a single `AddEscrow` event was emitted which did not correctly
+  represent the actual state changes when rewards were disbursed.
+
+  Now the following events are emitted:
+
+  - `Transfer(CommonPool -> Recipient, commissionAmount)` event for the
+    commissioned part of the reward
+  - `AddEscrow(Recipient -> Recipient, commissionAmount)` event for the
+    automatically escrowed commission reward
+  - `AddEscrow(CommonPool -> Recipient, restAmount)` for the non-commissioned
+    part of the reward (which only increases existing shares prices)
+
+### Configuration Changes
+
+- Change IAS proxy client authentication configuration
+  ([#3958](https://github.com/oasisprotocol/oasis-core/issues/3958))
+
+  Previously the full TLS certificate needed to be passed when configuring the
+  IAS proxy client, but now only the public key needs to be included.
+
+  The `ias.tls` configuration option is removed and `ias.proxy_addr` is
+  renamed to `ias.proxy.address`. One can now configure multiple proxy
+  addresses and each address needs to contain the Base64-encoded public key,
+  for example: `<pubkey>@ias.oasis.dev`.
+
+- go/worker/compute: remove `schedule_check_tx.enabled` flag
+  ([#3965](https://github.com/oasisprotocol/oasis-core/issues/3965))
+
+  Runtime executor workers now always call CheckTx before scheduling runtime
+  transactions. The option to disable CheckTx for runtime transactions is
+  removed.
+
+### Features
+
+- Bump Badger database to v3
+  ([#3859](https://github.com/oasisprotocol/oasis-core/issues/3859))
+
+  The old database will automatically be converted during the initial node
+  startup. For this reason the following should be taken into account when
+  upgrading:
+
+  - Initial startup time will be longer as a full database migration needs to
+    be performed for each of the internal databases. The migration may take up
+    to an hour for large databases.
```

Comment:
Did we ever test this on the mainnet db? Do we have an expected update time for there?

Reply:
Yes, it took a few minutes.

Reply:
Let's maybe update a note here that it should be few minutes for the mainnet? Or replace the "hour long for large databases" as it can be a bit misleading. I think most would classify the mainnet db as "large" :grin:

Reply:
Yeah, updated.